beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBatchFetcher.java

```diff
@@ -74,8 +75,9 @@ public class HistoricalBatchFetcher {
   private final BlobSidecarManager blobSidecarManager;
   private final SafeFuture<BeaconBlockSummary> future = new SafeFuture<>();
   private final Deque<SignedBeaconBlock> blocksToImport = new ConcurrentLinkedDeque<>();
-  private final Map<UInt64, List<BlobSidecar>> blobSidecarsBySlotToImport =
+  private final Map<SlotAndBlockRoot, List<BlobSidecar>> blobSidecarsBySlotToImport =
       new ConcurrentHashMap<>();
+  private Optional<UInt64> maybeEarliestBlobSidecarSlot = Optional.empty();
```

Comment:
better to put it as `volatile`? can different threads access this non-final field?

Reply:
I think if `alwaysRun` fragment is executed in the same time (and it's the only other usage) when we do import, we definitely have problems in the whole class. I'm not sure we need volatile here